tests/integration_tests/functional/test_api.py

```diff
@@ -37,6 +38,67 @@ def test_api_happy_start(test_microvm_with_api):
     test_microvm.start()
 
 
+def test_drive_io_engine(test_microvm_with_api, network_config):
+    """
+    Test io_engine configuration.
+
+    Test that the io_engine can be configured via the API on kernels that
+    support the given type and that FC returns an error otherwise.
+
+    @type: functional
+    """
+    test_microvm = test_microvm_with_api
+    test_microvm.spawn()
+
+    test_microvm.basic_config(add_root_device=False)
+    test_microvm.ssh_network_config(network_config, '1')
+
+    response = test_microvm.drive.put(
+        drive_id='rootfs',
+        path_on_host=test_microvm.create_jailed_resource(
+            test_microvm.rootfs_file),
+        is_root_device=True,
+        is_read_only=False,
+        # Set the opposite of the default backend type.
+        io_engine="Sync" if compare_versions(
+            get_kernel_version(), "5.10.0") > 0 else "Async"
+    )
+
+    if compare_versions(get_kernel_version(), "5.10.0") < 0:
```

Comment:
This is increasingly more used lately. I think helper methods would be useful, similar to what we have in the rust code. Something like `is_io_uring_supported`.